### PR TITLE
Hi scraper #371

### PR DESCRIPTION
**Pull request by Ash1R**
For issue[ #371](https://github.com/biglocalnews/warn-scraper/issues/371), added a scraper for HI data. 
It was a relatively simple list of pdf links, the instructions were to simply get the links since the format inside was too irregular.

### CODE REVIEW

File: warn/scrapers/mi.py [hunk 1]
```diff
@@ -1,6 +1,7 @@
 import re
```
Comment (palewire) [inline warn/scrapers/mi.py:1]: The Michigan scraper shouldn't be in the HI PR. Can we pull this out?

File: warn/scrapers/wv.py [hunk 2]
```diff
@@ -0,0 +1,90 @@
+from pathlib import Path
```
Comment (palewire) [inline warn/scrapers/wv.py:1]: The WV scraper shouldn't be in the HI PR. Can we pull this out?

**Review by palewire — CHANGES_REQUESTED** [review]

File: warn/scrapers/mi.py (hunk 1 above)
Comment (Ash1R) [reply warn/scrapers/mi.py:1]: Sure!

File: warn/scrapers/wv.py (hunk 2 above)
Comment (Ash1R) [reply warn/scrapers/wv.py:1]: Sure!

File: warn/scrapers/mi.py (hunk 1 above)
Comment (palewire) [reply warn/scrapers/mi.py]: The Michigan scraper is now being deleted. Isn't it already in the main branch? If so, we don't want to remove it here. Can you get it back in, unedited?

File: warn/scrapers/hi.py [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+                    )
+                    includes_date = True
+                elif j in dates[i].split(" ")[-3]:
+                    dates[i] = (
+                        dates[i].split(" ")[-3]
+                        + " "
+                        + dates[i].split(" ")[-2]
+                        + " "
+                        + dates[i].split(" ")[-1]
+                    )
```
Comment (palewire) [inline warn/scrapers/hi.py:69]: Can you explain to me why we need all this date parsing? Is there a reason we can't pass the dates into [strptime](https://docs.python.org/3/library/datetime.html#strftime-and-strptime-behavior) or [dateutil](https://pypi.org/project/python-dateutil/) or another date parser?

Comment (Ash1R) [reply warn/scrapers/hi.py:69]: Some entries have dates, but other entries are corrections which don't have dates. 
Some entries have the date at the end, and others at the beginning, hence the if and elif near the bottom. I extracted the parts of the entry which contain the date and put them in a string. 

Can dateutil or strptime read the whole entry and extract the date?

Comment (palewire) [reply warn/scrapers/hi.py:69]: I think it might. Give it a try.

Comment (Ash1R) [reply warn/scrapers/hi.py:69]: Gotta love fuzzy parsing, it works. 
However, 
Screen Shot 2023-01-18 at 12 51 05 AM
When I try to commit, it says that "library stubs" are not installed for dateutil or dateutil.parser. I tried running the suggested commands, but I'm still getting the same error.

Comment (palewire) [reply warn/scrapers/hi.py:69]: Try running adding `types-python-dateutil` here: https://github.com/biglocalnews/warn-scraper/blob/main/.pre-commit-config.yaml#L51-L53

Comment (Ash1R) [reply warn/scrapers/hi.py]: The error is gone and the code is pushed, thanks!

File: .pre-commit-config.yaml [hunk 4]
```diff
@@ -1,52 +1,54 @@
 # See https://pre-commit.com for more information
 # See https://pre-commit.com/hooks.html for more hooks
 repos:
```
Comment (palewire) [inline .pre-commit-config.yaml:3]: Can you please remove the changes to this file from the PR?

**Review by palewire — DISMISSED** [review]

File: .pre-commit-config.yaml (hunk 4 above)
Comment (Ash1R) [reply .pre-commit-config.yaml]: There was no option to comment on this til now.... but I fixed a few weeks ago.

**Comment by stucka** [comment]
Going to see if we can get a fresh code review by closing and reopening.

**Comment by stucka** [comment]
Looks like this may require a few more things to clear builds and tests:
types-python-dateutil = "*"
python-dateutil = "*"

I'll plan to evaluate it and see if dateutil is actually needed or if we can work without it.

pipenv generally does not like me adding stuff, and this rolled in with 300 lines deleted and 300 lines added.

**Comment by stucka** [comment]
Good and bad news here:

- The scraper remains effective when pointed at a proper web page.
- The web pages are now organized on a per-year basis, requiring some looping and additional scrapes
- There's just not much data that's directly scrapable
- I suspect I can get rid of the dateutil requirement here

**Comment by stucka** [comment]
@Ash1R , can you review?

**Comment by stucka** [comment]
Linting changes seem to have stuck this time. I'm backing away slowly without making eye contact.

**Review by Ash1R — COMMENTED** [review]
Looks good to me.. I don't see any issues and it seems a lot less vulnerable to edge cases.

**Comment by stucka** [comment]
Thank you, @Ash1R . And thank you!

**Pull request merged**